Tweak shadow bias defaults for DirectionalLight3D and OmniLight3D

Calinou opened the pull request:
Follow-up to https://github.com/godotengine/godot/pull/51335.

Light3D now uses a normal bias of 2 (instead of 1) to better combat shadow acne, without introducing more peter-panning.

The new values behave fairly well at increased blur values (between 1 and 2.5), but extreme blur values still don't look good out of the box. We can further improve reduce acne/peter-panning at high blur values by multiplying the shadow bias by the blur value (when it's above 1), but I think this is better to do in a separate PR.
**Edit:** Done in https://github.com/godotengine/godot/pull/55758 :slightly_smiling_face: 

This closes https://github.com/godotengine/godot/issues/54523.

**Testing project:** [test_shadows.zip](https://github.com/godotengine/godot/files/7686674/test_shadows.zip)

## Preview

*Click to view at full size.*

***Note:** Debanding is enabled to ensure that color banding doesn't get confused for shadow acne. Lights also have their Specular property set to 0.0.*

### DirectionalLight3D

| Before | After |
|-|-|
| ![2021-12-09_16 42 20](https://user-images.githubusercontent.com/180032/145432269-eac13b41-27b3-417c-a188-02f3f8d6c1a7.png) | ![2021-12-09_16 42 35](https://user-images.githubusercontent.com/180032/145432278-840d97b0-770a-4406-8a75-b3d041f9cad0.png) |

### OmniLight3D

| Before | After |
|-|-|
| ![2021-12-09_16 44 07](https://user-images.githubusercontent.com/180032/145432280-9cad5601-b576-4771-802c-ed8d807500fe.png) | ![2021-12-09_16 44 16](https://user-images.githubusercontent.com/180032/145432284-46c960a2-bcda-4f0b-b9b1-2414c2cff67f.png) |

mrjustaguy commented:
Do note that when using blending, higher Normal Bias will result in a more clearly visible Blend-fade effect when transitioning between splits.

mrjustaguy commented:
Maybe Normal Bias should be set to 2 for Omni lights too, as this seems to very well clear out Shadow Acne, and doesn't really seem to have any clear downsides compared to their current default. Spot Lights should remain as they are, as they don't benefit from this, and do actually suffer from adverse side-effects

Here is an Omni Light for comparison, Ultra Soft Shadows with 2048 General Shadows and quadrant used is split into 4 shadows, using 0.03 bias
Normal Bias 1
![N1](https://user-images.githubusercontent.com/50302479/152572871-54c92128-df32-4799-a117-2af5459a55c5.png)
Normal Bias 2
![N2](https://user-images.githubusercontent.com/50302479/152572886-059cb49e-6fe3-4cd2-a468-88b6d1df05d8.png)

mrjustaguy commented:
Omni Lights should not have such a low bias, as it creates acne for larger lights, I've found 0.1 bias and 2 normal bias works best for them (even with Complex Objects such as Humans casting shadows)

Small Light (5 radius, 1 Energy, Proposed Settings)
![image](https://user-images.githubusercontent.com/50302479/159007002-73681af1-2a23-430d-83a1-03f359e3515c.png)

Moderate Light (50 radius, 10 Energy, Proposed Settings)
![image](https://user-images.githubusercontent.com/50302479/159007574-edfaa6d0-c797-4cc6-ab38-b7566c10dd8f.png)

Large Light (500 radius, 100 Energy, Proposed Settings) Note some acne, but users would generally have to increase the Bias for larger lights anyway... Might be doable automatically? in which case 0.03 for 5 makes sense, but how to get to larger lights and keep them acne free?
![image](https://user-images.githubusercontent.com/50302479/159007786-f5a6c4c4-4b8d-4933-a208-6eef0ad826db.png)

Here's Moderate Light for comparison with This PR's settings, and Normal Bias at 2 Doesn't Help, and at Higher it just makes a ton of other problems
![image](https://user-images.githubusercontent.com/50302479/159007959-bfdbadc8-d87d-47d8-9734-7fbabb163610.png)

Calinou commented:
Redone to only increase Normal Bias. Bias is no longer changed by this PR.

reduz commented:
Bias needs to be different per light, so please make sure to test this well.

clayjohn reviewed (commented):
I ran your test project again to test these settings on Beta 7. I found that the SpotLight looked fine as it was. The Omnilight indeed benefited from decreasing shadow_bias, but it did not benefit from increasing the normal_bias. DirectionalLight3D did benefit from increasing normal_bias to 2 and it is fine with its current shadow bias of 1.0.

All in, what appeared to be the ideal settings are:
- SpotLight3D: leave as is
- Omnilight3D: decrease shadow bias to 0.1
- DirectionalLight3D: increase normal_bias to 2

Calinou committed:
Tweak shadow bias defaults for DirectionalLight3D and OmniLight3D

- Increase DirectionalLight3D normal bias to 2.0 to reduce shadow acne
  at grazing angles.
- Decrease OmniLight3D bias to 0.1 to reduce shadow peter-panning.

clayjohn reviewed (approved):
Looks fine to me. Although I am confused why the docs indicate the Light3D normal bias is 2.0 (which should only be the case for DirectionalLight3D). You would think it would be reversed and the Light3D normal bias would be 1.0 and only DirectionalLight3D would override it to change it to 2.0

I guess this is an issue that can be investigated in the docs binding later.

akien-mga commented:
> Looks fine to me. Although I am confused why the docs indicate the Light3D normal bias is 2.0 (which should only be the case for DirectionalLight3D). You would think it would be reversed and the Light3D normal bias would be 1.0 and only DirectionalLight3D would override it to change it to 2.0
> 
> I guess this is an issue that can be investigated in the docs binding later.

`Light3D` is abstract so it can't be instantiated by the doc generator to get its default values. So they end up inferred from a derived class, and it seems to pick one relatively arbitrarily.

akien-mga commented:
Thanks!
